src/Shopify.php

```diff
@@ -218,40 +246,90 @@ public static function make($shop, $token)
     /**
      * Get a resource using the assigned endpoint ($this->endpoint).
      *
-     * @param array  $query
+     * @param array $query
      * @param string $append
      *
-     * @throws InvalidOrMissingEndpointException
-     *
      * @return array
+     * @throws GuzzleException
+     *
+     * @throws InvalidOrMissingEndpointException
      */
     public function get($query = [], $append = '')
     {
         $api = $this->api;
 
+        // Don't allow use of page query on cursored endpoints
+        if (isset($query['page']) && in_array($api, static::$cursored_enpoints, true)) {
+            Log::warning(__METHOD__.': Use of deprecated query parameter. Use cursor navigation instead.');
+
+            return [];
+        }
+
+        // Do request and store response in variable
         $response = $this->request(
             $method = 'GET',
             $uri = $this->uri($append),
             $options = ['query' => $query]
         );
 
+        // If response has Link header, parse it and set the cursors
+        if ($response->hasHeader('Link')) {
+            $this->cursors = static::parseLinkHeader($response->getHeader('Link')[0]);
+        }
+
         $data = json_decode($response->getBody()->getContents(), true);
 
-        if (isset($data[static::apiCollectionProperty($api)])) {
-            return $data[static::apiCollectionProperty($api)];
+        return $data[static::apiCollectionProperty($api)] ?? $data[static::apiEntityProperty($api)] ?? $data;
+    }
+
+    /**
+     * @param array $query
+     * @param string $append
+     *
+     * @return array|null
+     *
+     * @throws GuzzleException
+     * @throws InvalidOrMissingEndpointException
+     */
+    public function next($query = [], $append = '')
+    {
+        // Only allow use of next on cursored endpoints
+        if (! in_array($this->api, static::$cursored_enpoints, true)) {
+            Log::warning(__METHOD__.': Use of cursored method on non-cursored endpoint.');
+
+            return [];
         }
 
-        if (isset($data[static::apiEntityProperty($api)])) {
-            return $data[static::apiEntityProperty($api)];
+        // If cursors haven't been set, then just call get normally.
+        if (empty($this->cursors)) {
+            return $this->get($query, $append);
         }
 
-        return $data;
+        // Only limit key is allowed to exist with cursor based navigation
+        foreach (array_keys($query) as $key) {
+            if ($key !== 'limit') {
+                Log::warning(__METHOD__.': Use of unallowed query param with cursored navigation');
```

Comment:
Some people using this package will not be using Laravel.

Reply:
Maybe 

```
defined('LARAVEL_START')
    ? Log::warning();
    : // see `https://www.php.net/manual/en/function.error-log.php`
```

Also, it might be prettier to set a flag on the class for `$this->laravel` and add a `isLaravel()`